src/schema/posts.ts

```diff
@@ -859,6 +925,80 @@ export const resolvers: IResolvers<any, Context> = {
 
       return { _: true };
     },
+    editPost: async (
+      source,
+      args: EditPostArgs,
+      ctx: Context,
+    ): Promise<GQLEmptyResponse> => {
+      const { id, image } = args;
+      const { con, userId } = ctx;
+      const title = args.title?.trim() ?? '';
+      const content = args.content?.trim() ?? '';
+
+      await con.transaction(async (manager) => {
+        const repo = manager.getRepository(Post);
+        const post = (await repo.findOneByOrFail({ id })) as
+          | WelcomePost
+          | FreeformPost;
+
+        if (!editablePostTypes.includes(post.type)) {
+          throw new ForbiddenError(
+            'Editing post outside of type welcome_post and freeform is not allowed',
+          );
+        }
+
+        if (post.authorId !== userId) {
+          const permission =
+            post.type === PostType.Welcome
+              ? SourcePermissions.WelcomePostEdit
+              : SourcePermissions.Post;
+          await ensureSourcePermissions(ctx, post.sourceId, permission);
+        }
+
+        if (title.length > MAX_TITLE_LENGTH) {
+          throw new ValidationError(
+            'Title has a maximum length of 80 characters',
+          );
+        }
+
+        if (content.length > MAX_CONTENT_LENGTH) {
+          throw new ValidationError(
+            'Content has a maximum length of 80 characters',
+          );
+        }
+
+        const updated: Partial<EditablePost> = {};
+
+        if (post.type === PostType.Freeform) {
+          updated.title = title;
+
+          if (image && process.env.CLOUDINARY_URL) {
+            const upload = await image;
+            updated.image = await uploadPostImage(
+              id,
+              upload.createReadStream(),
+            );
+          }
+        }
+
+        const mentions = await getMentions(
```

Comment:
Shouldn't this be inside the if block?

Reply:
Ah, right. I forgot we are doing a transaction, so even if I save the mentions first on the code they will still run at the same time.